crates/wit-bindgen-wasmcloud/src/lib.rs

```diff
@@ -0,0 +1,5 @@
+pub mod provider {
+    pub mod binary {
+        pub use wit_bindgen_wasmcloud_provider_binary::generate;
+    }
+}
```

Comment:
Could you help me understand why all the nesting here is necessary?

Reply:
Since there are different types of providers (rust-only, wasm guest, wasm host if we ever do that) I prefer `wit-bindgen-wasmcloud::provider::<type>::generate`

Reply:
@vados-cosmonic I think we're only going to ever have binary provider gen stuff until we move to normal wasi providers, which will use the open source toolchain (with some of the modifications or similar to what you added). Let's just call this `wit_bindgen_wasmcloud_provider` and not have any nesting for now. If we do end up with more stuff we can nest then

Reply:
OK, sure -- will change this to `wit_bindgen_wasmcloud_provider`, and remove the second project completely
- [ ] rename to `wit_bindgen_wasmcloud_provider`